storage/store.go

```diff
@@ -292,9 +292,17 @@ func NewStore(clock *hlc.Clock, eng engine.Engine, db *client.KV, gossip *gossip
 	// Add range scanner and configure with queues.
 	s.scanner = newRangeScanner(defaultScanInterval, newStoreRangeIterator(s))
 	s.gcQueue = newGCQueue()
-	s.splitQueue = newSplitQueue(db, gossip)
 	s.verifyQueue = newVerifyQueue(s.scanner.Stats)
-	s.scanner.AddQueues(s.gcQueue, s.splitQueue, s.verifyQueue)
+	s.scanner.AddQueues(s.gcQueue, s.verifyQueue)
+	if db != nil {
```

Comment:
Instead of putting all of this logic here, why not just add something to the split queue which checks whether db is nil as part of ShouldQueue method. Print a warning if nil and return false.